packages/grid/x-data-grid/src/hooks/core/preProcessing/gridPreProcessingApi.ts

```diff
@@ -54,14 +49,14 @@ export interface GridPreProcessingApi {
    * Register a pre-processor and emit an event to notify the agents to re-apply the pre-processors.
    * @param {GridPreProcessingGroup} group The name of the group to bind this pre-processor to.
    * @param {number} id An unique and static identifier of the pre-processor.
-   * @param {PreProcessorCallback} callback The pre-processor to register.
+   * @param {GridPreProcessor} callback The pre-processor to register.
    * @returns {() => void} A function to unregister the pre-processor.
    * @ignore - do not document.
    */
-  unstable_registerPreProcessor: (
+  unstable_registerPreProcessor: <G extends GridPreProcessingGroup>(
```

Comment:
I improved the typing of the pipe processors to match the strategy processors format